### PR TITLE
category

### PR DESCRIPTION
**Pull request by nguyenngoctruong**

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class Category < ActiveRecord::Base
+	# has_may :words
+	validates :name, presence: true, length: { maximum: 200 }
```
Comment (nguyenduyta) [inline app/models/category.rb]: Coding convention

File: app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    respond_to do |format|
+      format.html {redirect_to categories_path}
+      format.js
+    end
+  end
+
+  private
+
+  def category_params
+    params.require(:category).permit(:name)
```
Comment (dieunb) [inline app/controllers/categories_controller.rb]: remove `()` at last

File: app/views/categories/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<% provide(:title, 'New category') %>
+<h1>New category</h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@category) do |f| %>
+      <%= f.label :name %>
```
Comment (dieunb) [inline app/views/categories/new.html.erb]: error message???

File: app/models/category.rb [hunk 4]
```diff
@@ -1,2 +1,4 @@
 class Category < ActiveRecord::Base
+  # has_may :words
+  validates :name, presence: true, length: { maximum: 200 }
```
Comment (dieunb) [inline app/models/category.rb]: coding convention

File: app/views/categories/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<% provide(:title, 'All Categories') %>
```
Comment (dieunb) [inline app/views/categories/index.html.erb]: remove `()`

File: app/views/categories/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+<h1>All Categories</h1>
+<ul class="categories" id="categories">    
+  <%= render @categories %>
+</ul>
+<div class="center"><%= link_to('New Category', new_category_path, class: "btn btn-primary") %></div>
```
Comment (dieunb) [inline app/views/categories/index.html.erb]: remove `()`

File: app/views/categories/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<% provide(:title, 'New category') %>
```
Comment (nguyenduyta) [inline app/views/categories/new.html.erb]: remove () =))

**Comment by dieunb** [comment]
OK

**Pull request merged**